include/cell.tcc

```diff
@@ -856,6 +856,58 @@ Eigen::VectorXd mpm::Cell<Tdim>::compute_strain_rate(
   return strain_rate;
 }
 
+//! Compute strain rate for reduced integration at the centroid of cell
+template <unsigned Tdim>
+Eigen::VectorXd mpm::Cell<Tdim>::compute_strain_rate_centroid(unsigned phase) {
+  // Define strain rate
+  Eigen::VectorXd strain_rate;
+
+  switch (Tdim) {
+    case (1): {
+      strain_rate.resize(1);
+      break;
+    }
+    case (2): {
+      strain_rate.resize(3);
+      break;
+    }
+    default: {
+      strain_rate.resize(6);
+      break;
+    }
+  }
+
+  strain_rate.setZero();
+
+  // Get centroid local coordinates, which are zeros
+  Eigen::Matrix<double, Tdim, 1> xi_centroid;
+  xi_centroid.setZero();
+
+  // Initialize B-Matrix
+  std::vector<Eigen::MatrixXd> bmatrix;
+
+  try {
+    // Get B-Matrix at the centroid
+    bmatrix = element_->bmatrix(xi_centroid);
```

Comment:
Is this the right B-Matrix, this only gives the B-matrix for an element in natural coordinates

Reply:
Yes, thus the centroid is always xi = (0, 0, 0)

Reply:
That only fixes the geometry, don't you need dN/dX and not dN/dxi

Reply:
Understood... let me do some calculations for it.